cmd/runtimecfg/node-ip.go

```diff
@@ -91,47 +94,68 @@ func set(cmd *cobra.Command, args []string) error {
 	}
 	log.Infof("Chosen Node IPs: %v", chosenAddresses)
 
-	// Kubelet
-	kubeletOverrideDir := filepath.Dir(kubeletSvcOverridePath)
-	err = os.MkdirAll(kubeletOverrideDir, 0755)
-	if err != nil {
-		return err
-	}
-	log.Infof("Opening Kubelet service override path %s", kubeletSvcOverridePath)
-	kOverride, err := os.Create(kubeletSvcOverridePath)
-	if err != nil {
-		return err
-	}
-	defer kOverride.Close()
-
 	nodeIP := chosenAddresses[0].String()
 	nodeIPs := nodeIP
 	if len(chosenAddresses) > 1 {
 		nodeIPs += "," + chosenAddresses[1].String()
 	}
+	// Kubelet
 	kOverrideContent := fmt.Sprintf("[Service]\nEnvironment=\"KUBELET_NODE_IP=%s\" \"KUBELET_NODE_IPS=%s\"\n", nodeIP, nodeIPs)
 	log.Infof("Writing Kubelet service override with content %s", kOverrideContent)
-	_, err = kOverride.WriteString(kOverrideContent)
+	err = writeToFile(kubeletSvcOverridePath, kOverrideContent)
 	if err != nil {
 		return err
 	}
 
 	// CRI-O
-	crioOverrideDir := filepath.Dir(crioSvcOverridePath)
-	err = os.MkdirAll(crioOverrideDir, 0755)
+	cOverrideContent := fmt.Sprintf("[Service]\nEnvironment=\"CONTAINER_STREAM_ADDRESS=%s\"\n", nodeIP)
+	log.Infof("Writing CRIO service override with content %s", cOverrideContent)
+	err = writeToFile(crioSvcOverridePath, cOverrideContent)
+	if err != nil {
+		return err
+	}
+
+	// node ip hint for all other services
+	err = writeToFile(nodeIpFile, nodeIP)
+	if err != nil {
+		return err
+	}
+
+	ipv6Created, ipv4Created := false, false
+	for i := 0; i < len(chosenAddresses) && i < 2; i++ {
+		if utils.IsIPv6(chosenAddresses[i]) && !ipv6Created {
+			err = writeToFile(nodeIpIpV6File, chosenAddresses[i].String())
+			if err != nil {
+				return err
+			}
+			ipv6Created = true
+		} else if !ipv4Created {
```

Comment:
This makes me a little nervous. If we ever got two ipv6 chosenAddresses, it could potentially result in us writing an ipv6 address to the ipv4 file. To avoid checking IsIPv6 twice (which, honestly, I think is premature optimization. This is a service that runs once per boot and is not performance critical code), I think we need that to be an outer check and then within the cases for it we do the ipv*Created checks. Something like:

```
if isipv6 {
  if !ipv6Created {
    create file
  }
} else {
  if !ipv4Created {
     create file
  }
}
```

Note that I don't think it's currently possible to get two ipv6 addresses in chosenAddresses, but if I'm wrong or that changes in the future I'd prefer to have this code handle it correctly.

Reply:
This is actually a bug from my side . We should check if nit ipv6 , this is why it was there from the start. Sorry